base/acme/src/main/java/org/dogtagpki/acme/issuer/PKIIssuer.java

```diff
@@ -77,7 +80,40 @@ public void init() throws Exception {
             clientConfig.setUsername(username);
         }
 
-        String password = config.getParameter("password");
+        String password = null;
+        String passwordFilename = config.getParameter("passwordFile");
+        if (passwordFilename != null) {
+            // read password from file
+            File passwordFile = new File(passwordFilename);
+
+            if (passwordFile.canRead()) {
```

Comment:
I don't think we need to call `File.canRead()` explicitly. If the file is missing or cannot be read, an exception will get thrown automatically when the code tries to read the file. It may even provide a more specific exception type or error message.